### PR TITLE
[codemirror] add `configureMouse` to EditorConfiguration

### PR DESCRIPTION
**Pull request by lenardfunk**
Added type description for `EditorConfiguration.configureMouse` to types/codemirror/index.d.ts, which is available since CodeMirror [v5.27.0](https://github.com/codemirror/CodeMirror/blob/master/CHANGELOG.md#new-features-31).

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: 
   - Docs: https://codemirror.net/doc/manual.html#option_configureMouse
   - Code: https://github.com/codemirror/CodeMirror/blob/master/src/edit/mouse_events.js#L116
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## codemirror ([<kbd>unpkg</kbd>](https://unpkg.com/browse/codemirror@latest/))
was missing the following properties:
1. optionHandlers
1. inputStyles
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e35a1c2f4a0be28a273802eef41a639174e78d32
</p>

File: types/codemirror/index.d.ts [hunk 1]
```diff
@@ -1256,6 +1256,13 @@ declare namespace CodeMirror {
 
         /** Optional lint configuration to be used in conjunction with CodeMirror's linter addon. */
         lint?: boolean | LintStateOptions | Linter | AsyncLinter;
+
+        /** Allows you to configure the behavior of mouse selection and dragging. The function is called when the left mouse button is pressed. */
+        configureMouse?: (
```
Comment (ficristo) [inline types/codemirror/index.d.ts:1261]: Could this moved between `extraKeys` and `lineWrapping` to match the position in the manual documentation?

File: types/codemirror/index.d.ts [hunk 2]
```diff
@@ -1888,4 +1895,32 @@ declare namespace CodeMirror {
             setShowDifferences(showDifferences: boolean): void;
         }
     }
+
+    interface MouseSelectionConfiguration {
```
Comment (ficristo) [inline types/codemirror/index.d.ts:1899]: I think all the properties are optional so they should be something like `unit?: ..., extend?: ...`

**Review by ficristo — CHANGES_REQUESTED** [review]

File: types/codemirror/index.d.ts (hunk 2 above)
Comment (lenardfunk) [reply types/codemirror/index.d.ts:1899]: That's true, thank you!

File: types/codemirror/index.d.ts (hunk 1 above)
Comment (lenardfunk) [reply types/codemirror/index.d.ts]: Done!

**Review by ficristo — APPROVED** [review]

**Comment by lenardfunk** [comment]
Ready to merge

**Pull request merged**